dev-infrastructure/templates/dev-aks-roleassignments.bicep

```diff
@@ -0,0 +1,28 @@
+// This is used to grant CI the ability to deploy resources into
+// dev AKS clusters. It should not be used in higher environments.
+param aksClusterName string
+param location string
+
+var githubActionsAppID = '3aa9b94b-acfc-4e7e-b952-158a174cb1bb'
```

Comment:
Is this the application id, or the principal id of the service principal?  RBAC is defined on the latter.  

We should probably drop it as a secret into the GH repo as well, or we can query it using MSGraph using the application id.

Reply:
ack, I tested the RBAC grant works, so it must be the principal ID. Bicep has a preview feature for MSGraph, so we can try that as well: https://learn.microsoft.com/en-us/graph/templates/overview-bicep-templates-for-graph

Reply:
Based on [THIS](https://learn.microsoft.com/en-us/graph/templates/quickstart-create-bicep-interactive-mode?tabs=CLI) using Graph with bicep is a preview feature, assuming we cant control that with github actions? For now ill update the variable name so it reflects the principal and add definitions to manifest, but someone would need to add the secret for me

Reply:
changed to a param called `githubActionsPrincipalID`, this way we can define it using `--parameters` and the github secret value